api4/apitestlib.go

```diff
@@ -392,12 +392,16 @@ func (me *TestHelper) CreateMessagePostWithClient(client *model.Client4, channel
 }
 
 func (me *TestHelper) CreateMessagePostNoClient(channel *model.Channel, message string, createAtTime int64) *model.Post {
-	post := store.Must(me.App.Srv.Store.Post().Save(&model.Post{
+	post, err := me.App.Srv.Store.Post().Save(&model.Post{
 		UserId:    me.BasicUser.Id,
 		ChannelId: channel.Id,
 		Message:   message,
 		CreateAt:  createAtTime,
-	})).(*model.Post)
+	})
+
+	if err != nil {
+		panic(err)
```

Comment:
I have a question here. In my understanding, we use panic in golang for unrecoverable error or a programmer error. Is it a good practice to use panic here or maybe just return the error message on failing to save a post?

Reply:
You are correct, best Golang practise is to return the error and handle the error type. But the code you are referring to is helper methods for creating `Posts`, the pattern for handling errors in `apitestlib.go` is to call `panic` when an error occurs. So I'm keeping the same pattern, so I don't have to change any return types and therefore not breaking the contract with the caller to this function.

But yeah, it can be changed to return the error instead. But not sure if it is part of this story and I also do not know the reasoning behind using panic in `apitestlib.go`.

But thanks for pointing it out 👍

Reply:
These functions are actually only used for testing, so we aren't as strict with following good practices here